utils.php

```diff
@@ -135,7 +135,7 @@ function touch_user($username)
 
 function add_user($username)
 {
-    if($user = get_user($username)) 
+    if($username == get_user($username)) 
```

Comment:
Sunt doua chichite aici: prima e ca faceai atribuire in if si am modificat-o intr-o verificare simpla  (stiu ca nu e cea mai buna idee, dar e functionala) si a doua e ca eu preupun ca era un typo din partea ta. Daca il lasi cum era inainte, o sa vezi ca nu intra in conditie (verificare contra unei variabile neinitializare) si ca atare nu scrie nimic in baza de date.